src/GraphQl/Resolver/ItemResolver.php

```diff
@@ -62,7 +69,15 @@ public function __invoke($source, $args, $context, ResolveInfo $info)
 
         $baseNormalizationContext = ['attributes' => $this->fieldsToAttributes($info)];
         try {
+            if (null !== $this->dispatcher) {
+                $this->dispatcher->dispatch(PreReadEvent::NAME, new PreReadEvent(null));
+            }
+
             $item = $this->iriConverter->getItemFromIri($args['id'], $baseNormalizationContext);
+
+            if (null !== $this->dispatcher) {
+                $this->dispatcher->dispatch(PostReadEvent::NAME, new PostReadEvent($item));
+            }
         } catch (ItemNotFoundException $e) {
             return null;
```

Comment:
wouldn't it be better to change this block to 
```php
try {
    if (null !== $this->dispatcher) {
        $this->dispatcher->dispatch(PreReadEvent::NAME, new PreReadEvent(null));
    }
    $item = $this->iriConverter->getItemFromIri($args['id'], $baseNormalizationContext);
} catch (ItemNotFoundException $e) {
    $item = null;
}

if (null !== $this->dispatcher) {
    $this->dispatcher->dispatch(PostReadEvent::NAME, new PostReadEvent($item));
}

if (null === $item) {
    return null;
}
```
to ensure that the PostReadEvent is always sent like in the ReadListener ?

Reply:
I don't think it should be dispatched when Read has failed.